.github/workflows/ci-sanitizers.yml

```diff
@@ -0,0 +1,128 @@
+name: CI with sanitizers enabled
+
+on:
+  # TODO: Enable pull request and cron triggers later, when this workflow is tested and not failing
+  # pull_request:
+  # schedule:
+    # - cron: '0 20 * * *' # Every day at 12pm PST (UTC-8)
+  workflow_dispatch:
+
+env:
+  BUILD_TYPE: Debug
+  MOUNT: /azure-osconfig
+  REGISTRY: ghcr.io
+  SANITIZER_FLAGS: -fsanitize=address,undefined -fno-omit-frame-pointer -g -O1
+
+jobs:
+  create-ci-matrix:
+    name: Build CI distro matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.get-matrix.outputs.matrix }}
+    steps:
+      - name: Get matrix
+        id: get-matrix
+        # Skip centos-7, oraclelinux-7, rhel-7 as they don't currently build properly from Dockerfile
+        run: |
+          matrix="$(cat <<'EOL'
+          [
+            { "name": "ubuntu-22.04", "arch": "amd64", "tag": "latest" }
```

Comment:
>{ "name": "ubuntu-22.04", "arch": "amd64", "tag": "latest" } [](http://example.com/codeflow?start=12&length=60)

Not for this PR now, but to consider for the future. We build OSConfig for lots more distros. There are differences among them for the code that is executed on each, esp. for functional tests. Also, AMD64-only is only for the MC scenarios. For OSConfig in general we also plan to re-add ARM64 in the future and in the past we also supported x86 and ARMHF/32 bit.

Reply:
I initially intended to include it on all platforms. but not all the images build now and fixing that is beyound this PR scope.